### PR TITLE
[Feature] Initialize shop service

### PR DESCRIPTION
**Pull request by natTP**
- Setup protobuf for `shop` service
- Setup gRPC server and client for `shop` service
- Add fields to shop migration 
- Add postgres models for shop and shopItem

### CODE REVIEW

File: lib/postgres/models/shop.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package models
+
+type Shop struct {
```
Comment (bombnp) [inline lib/postgres/models/shop.go:3]: GORM defaults to plural table name. To force table name to singular, u need to add a method to the model
```go
func (Shop) TableName() string {
	return "shop"
}
```

**Review by saenyakorn — COMMENTED** [review]
LGTM

**Review by saenyakorn — APPROVED** [review]

File: lib/postgres/models/shopItem.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package models
+
+type ShopItem struct {
```
Comment (bombnp) [inline lib/postgres/models/shopItem.go:3]: table name issue, same as shop

File: shop/internal/libproto/shop.proto [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    rpc GetAllShopItems(GetAllShopItemsRequest) returns (AllShopItemsResponse);
+    rpc GetShopItem(GetShopItemRequest) returns (ShopItemResponse);
+    rpc EditShopItem(EditShopItemRequest) returns (ShopItemResponse);
+    rpc DeleteShopItem(DeleteShopItemRequest) returns (ShopItemResponse);
+
+    rpc AddBankAccount(AddBankAccountRequest) returns (AddBankAccountResponse);
+}
+
+message Shop {
+    string name = 2;
```
Comment (bombnp) [inline shop/internal/libproto/shop.proto]: why start with 2?

File: shop/internal/libproto/shop.proto [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+message DeleteShopRequest {
+    string id = 1;
+}
+
+message ShopResponse {
+    Shop shop = 1;
+}
+
+message CreateShopItemRequest {
+    ShopItem item = 2;
```
Comment (bombnp) [inline shop/internal/libproto/shop.proto]: why start with 2?

File: shop/internal/libproto/shop.proto [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+message AllShopItemsResponse {
+    repeated ShopItem items = 1;
+}
+
+message AddBankAccountRequest {
+}
+
+message AddBankAccountResponse {
+}
```
Comment (bombnp) [inline shop/internal/libproto/shop.proto]: add newline at EOF

**Review by bombnp — CHANGES_REQUESTED** [review]

File: shop/internal/libproto/shop.proto (hunk 3 above)
Comment (natTP) [reply shop/internal/libproto/shop.proto]: forgot to change when moving `id`outside

**Pull request merged**